test/powershell/Language/LanguageTestSupport.psm1

```diff
@@ -65,9 +65,26 @@ function ShouldBeParseError
         # This is a temporary solution after moving type creation from parse time to runtime
         [switch]$SkipAndCheckRuntimeError,
         # for test coverarage purpose, tests validate columnNumber or offset
-        [switch]$CheckColumnNumber
+        [switch]$CheckColumnNumber,
+        # Skip this test in Travis CI nightly build
+        [switch]$SkipInTravisFullBuild
     )
 
+    #
+    # CrossGen'ed assemblies cause a hang to happen when running tests with this helper function in Linux and OSX.
+    # The issue has been reported to CoreCLR team. We need to work around it for now with the following approach:
+    #  1. For pull request and push commit, build without '-CrossGen' and run the parsing tests
+    #  2. For nightly build, build with '-CrossGen' but don't run the parsing tests
+    # In this way, we will continue to exercise these parsing tests for each CI build, and skip them for nightly
+    # build to avoid a hang.
+    # Note: this change should be reverted once the 'CrossGen' issue is fixed by CoreCLR. The issue is tracked by
+    #       https://github.com/dotnet/coreclr/issues/9745
+    #
+    if ($SkipInTravisFullBuild) {

```

Comment:
maybe add a comment here to the effect of 'report that we skipped the test and return'